core/server/data/migrations/versions/2.37/01-add-self-signup-and-from address-to-members-settings.js

```diff
@@ -0,0 +1,71 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const common = require('../../../../lib/common');
+
+module.exports.config = {
+    transaction: true
+};
+
+module.exports.up = (options) => {
+    let localOptions = _.merge({
+        context: {internal: true}
+    }, options);
+    const settingsKey = 'members_subscription_settings';
+    return localOptions
+        .transacting('settings')
+        .then((response) => {
+            if (!response) {
+                common.logging.warn('Cannot find settings.');
+                return;
+            }
+            let subscriptionSettingsEntry = response.find((entry) => {
+                return entry.key === settingsKey;
+            });
+
+            if (!subscriptionSettingsEntry) {
+                common.logging.warn('Cannot find members subscription settings.');
+                return;
+            }
+            let subscriptionSettings = JSON.parse(subscriptionSettingsEntry.value);
+
+            let hasRequirePaymentProperty = Object.prototype.hasOwnProperty.call(subscriptionSettings, 'requirePaymentForSignup');
+            let hasSelfSignupProperty = Object.prototype.hasOwnProperty.call(subscriptionSettings, 'allowSelfSignup');
+            let hasFromAddressProperty = Object.prototype.hasOwnProperty.call(subscriptionSettings, 'fromAddress');
+            if (!hasRequirePaymentProperty && hasSelfSignupProperty) {
+                common.logging.info(`Skip setting allowSelfSignup property on member settings because it already exists`);
+                return Promise.resolve();
```

Comment:
This doesn't take into account that `hasFromAddressProperty` might be false and would not set `fromAddress` to `'noreply'` - should we cover that case?

Reply:
Yes think we should, I had included `fromAddress` slightly later so def just a case of missing it. 👍